### ISSUE TITLE
How to install

### ISSUE THREAD

**Issue by akshitgaur2005** [idlip/d-nix#2]
Hi, I am a newbie to nixos though I have done some customizations and have been playing with it for about 2 weeks now. But I just can't seem to understand how to reproduce this build. If you could give some pointers it would be really awesome.

**Comment by idlip** [comment]
Hi @akshitgaur2005 !

Thank you for reaching out.

I'm not sure in what you mean to reproduce. Are you asking on how to use literate org config (with emacs) and have a nixos config?

Or do you just mean to get some of the tools configuration from my config?

Please do clarify more in details on what exact issue or query you are looking for.

I maintain all my config in a single file basis, so everything is either in `core.nix` for nixos configuration and modules & `home.nix` for home-manager modules.

If you are not targeting emacs org-mode based config, then https://github.com/Misterio77/nix-starter-configs might be a good start.

If you are looking for emacs org-mode based config, then to note and edit;
 I use `gdk` as my hostname and also as a directory to tangle files into. I use `idlip` as username. You may need to edit. Each src block is defined as a module under big list of `import` keyword. Thus with org, I can `COMMENT heading` or not tangle a certain blocks.
 
It's pretty straight forward as such with `sudo nixos-rebuild swtich --flake /path/directory#hostname` to build the system config.

Hope this helps, feel free to ask or have even more direct conversation via matrix or irc (optinally telegram) if you want.

**Comment by akshitgaur2005** [comment]
I was basically asking for how I could have the exact same system as you :sweat_smile: .
I actually have created my own basic config from the minimal starter config but I have no eye for the aesthetic stuff. While your config just look too damn good. So what I was hoping for is that I recreate your config and then change it according to my needs. But I have still much to learn in Nix and development in general. 
So if you could share a step-by-step guide to basically copying your system onto mine, it would be really appreciated.
Also, could you share your telegram username so that I can contact you there if you prefer?
Thanks and sorry for bothering you.

**Comment by idlip** [comment]
Okay I got it. Even I had the same feeling at some point, like let me just dive into it and later I will learn.

But I'd kindly like to warn you that aesthetic look does not really add much. You can watch many [r/unixporn](https://old.reddit.com/r/unixporn/) posts and search "Hyprland" or "nixos" to get many looks.

But honestly, all the aesthetics you see are only derived from "waybar" the status bar, and the WM "Hyprland".

Nowadays I feel like it is best to start from scratch, understand each parts and move according to the needs. I'm serious in this regards, even I was chasing behind looks, getting into trend. But using barebones, vanilla and understanding the components is really beneficial on long run.

Basically with my config, you can see each component/tools config and can copy each to your config and rebuild the system.

Even there is not much looks from my config now. I'm trying to go more minimal and vanilla.

Finally as I said, try with Hyprland, and some waybar configs, that is all the looks you are looking for.

Telegram id: ronin_zoro (you contact for more help with nixos)

Thank you

**Comment by akshitgaur2005** [comment]
Thanks for your detailed reply. I have actually done configuration and understand now, at least the classical hierarchical method of arranging the modules in folders. The most I learn is when I directly jump in so to speak actually and your org-mode method really caught my eye.
